crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -316,6 +317,11 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
         if status_reporter.errors_found() {
             errors_found = true;
         }
+
+        if let Err(err) = status_reporter.generate_final_result() {
+            log::error!("generate final result: {}", err);
+            return proc_exit::Code::FAILURE.ok();
```

Comment:
Why was the early return added?  Should we instead set `errors_found`?

Reply:
updated